packages/dev/docs/pages/blog/building-a-combobox.mdx

```diff
@@ -0,0 +1,127 @@
+since the tray can display more items on the screen and grants users a larger hit area to scroll through. To compensate for the fact that the tray covers the majority of the
+screen including the ComboBox, we included an input field within the tray so that end users could still type to filter the available items. So far so good. However,
+when we tested the ComboBox's tray input on iOS Safari, the tray's bottom half was covered by the onscreen keyboard and we could not scroll the hidden items into view.
+
+It turns out that iOS Safari does not shrink the browser window to accommodate the onscreen keyboard unlike other mobile browsers. Instead, Safari pushes the window upwards and
+partially off screen. At the time, we had been relying on `window.innerHeight` to inform us on how tall the tray should be but now we had to find a different way of tracking the available window
+space for the tray.
+
+Luckily for us, iOS 13 added support for the VisualViewport API. By querying `window.VisualViewport.height` we could get a reliable measurement of how much vertical space was available on screen. Furthermore, we could track
+when the onscreen keyboard was opened or dismissed by listening to the VisualViewport's `resize` event. Leveraging these two allowed us to create a tray that properly adjusts to the presence of iOS onscreen keyboard. Check out the video below
+to see how the ComboBox tray worked before and after we switched to the VisualViewport API. If you'd like to track the visual viewport size in your own app, you can use the `useViewportSize` hook available in the `@react-aria/utils` package.
+
+<Video src={comboboxVisualViewportUrl} loop autoPlay muted style={{maxWidth: 'min(30%, 350px)', display: 'block', margin: '20px auto'}} />
+
+### Page Scrolling
+
+Another issue we encountered had to do with iOS Safari page scrolling behavior. When the onscreen keyboard is visible, iOS Safari makes the page scrollable so that users can still access content that is hidden behind the keyboard. However,
+now that our ComboBox tray sizes itself to fit in the visual viewport, users could now scroll the entire tray itself off screen. To stop this from happening, we prevent default on `touchmove` events that happen on the document body
+or root element of the document. This preserves the user's ability to scroll through the options in the tray but blocks any attempt to scroll the page itself until the tray is closed. The video below
+illustrates the difference in scrolling behavior before and after our fix. If you are building your own overlays and would like to prevent this kind of document scrolling behavior, check out the [usePreventScroll](../react-aria/usePreventScroll.html) hook in the `react-aria/overlays` package.
+
+<Video src={comboboxScrollingUrl} loop autoPlay muted style={{maxWidth: 'min(30%, 350px)', display: 'block', margin: '20px auto'}} />
```

Comment:
Do we want a video or image showing it working after the changes?

Reply:
The video actually includes a before/after, just may not be super apparent